File: enrollment-system/src/main/java/org/app/enrollmentsystem/model/Course.java
package org.app.enrollmentsystem.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "course")
public class Course {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false, unique = true, length = 20)
  private String code;

  @Column(nullable = false, length = 100)
  private String name;

  @Column(columnDefinition = "TEXT")
  private String description;

  @Column(nullable = false)
  private Integer credits;

  @Column(nullable = false)
  private Integer semesterLevel;

  @CreationTimestamp
  @Column(nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
  private LocalDateTime registrationDate = LocalDateTime.now();

  @Column(nullable = false)
  private Boolean active = true;

  @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true)
  private List<Enrollment> enrollments;
}

File: enrollment-system-frontend/src/features/course/types/course.ts
export interface CourseResponse {
  id: number;
  code: string;
  name: string;
  description: string;
  credits: number;
  semesterLevel: number;
  active: boolean;
  registrationDate: string;
}

export type CourseRequest = Omit<
  CourseResponse,
  "id" | "active" | "registrationDate"
>;

export type CourseForm = {
  code: string;
  name: string;
  description?: string | "";
  credits: number;
  semesterLevel: number;
};
